experiments/ClimaEarth/components/land/climaland_integrated.jl

```diff
@@ -0,0 +1,540 @@
+    soilco2_bot_bc = CL.Soil.Biogeochemistry.SoilCO2FluxBC((p, t) -> 0.0) # no flux
+    soilco2_sources = (CL.Soil.Biogeochemistry.MicrobeProduction{FT}(),)
+    soilco2_boundary_conditions = (; top = soilco2_top_bc, bottom = soilco2_bot_bc)
+    return (;
+        boundary_conditions = soilco2_boundary_conditions,
+        sources = soilco2_sources,
+        domain = domain,
+        parameters = soilco2_ps,
+    )
+end
+
+"""
+    create_soil_args(::Type{FT}, domain, spatially_varying_soil_params)
+
+Creates the arguments for the soil model.
+"""
+function create_soil_args(::Type{FT}, domain, spatially_varying_soil_params) where {FT}
+    (;
+        ν,
+        ν_ss_om,
+        ν_ss_quartz,
+        ν_ss_gravel,
+        hydrology_cm,
+        K_sat,
+        S_s,
+        θ_r,
+        PAR_albedo_dry,
+        NIR_albedo_dry,
+        PAR_albedo_wet,
+        NIR_albedo_wet,
+        f_max,
+    ) = spatially_varying_soil_params
+
+    soil_params = CL.Soil.EnergyHydrologyParameters(
+        FT;
+        ν,
+        ν_ss_om,
+        ν_ss_quartz,
+        ν_ss_gravel,
+        hydrology_cm,
+        K_sat,
+        S_s,
+        θ_r,
+        PAR_albedo_dry = PAR_albedo_dry,
+        NIR_albedo_dry = NIR_albedo_dry,
+        PAR_albedo_wet = PAR_albedo_wet,
+        NIR_albedo_wet = NIR_albedo_wet,
+    )
+    return (domain = domain, parameters = soil_params)
+end
+
+"""
+    create_snow_args(::Type{FT}, domain, earth_param_set, dt)
+
+Creates the arguments for the snow model.
+"""
+function create_snow_args(::Type{FT}, domain, earth_param_set, dt) where {FT}
+    snow_parameters = CL.Snow.SnowParameters{FT}(dt; earth_param_set = earth_param_set)
+    return (; parameters = snow_parameters, domain = CL.obtain_surface_domain(domain))
+end
+
+###############################################################################
+### Functions required by ClimaCoupler.jl for a SurfaceModelSimulation
+###############################################################################
+
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:area_fraction}) = sim.area_fraction
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:beta}) =
+    CL.surface_evaporative_scaling(sim.model, sim.integrator.u, sim.integrator.p)
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:emissivity}) = sim.integrator.p.ϵ_sfc
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:energy}) = CL.total_energy(sim.integrator.u, sim.integrator.p)
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:surface_direct_albedo}) =
+    CL.surface_albedo(sim.model, sim.integrator.u, sim.integrator.p)
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:surface_diffuse_albedo}) =
+    CL.surface_albedo(sim.model, sim.integrator.u, sim.integrator.p)
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:water}) = CL.total_water(sim.integrator.u, sim.integrator.p)
+function Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:height_disp})
```

Comment:
I am not sure we need this function anymore

Reply:
Because the land knows how to access these things from itself (for surface fluxes), and we are computing surface fluxes with a land function

Reply:
I agree, but if in the future we want to compute fluxes by calling SurfaceFluxes directly from the coupler, maybe it would be good to know where to access these things in the land model. We can either keep it, or leave it out for now and add it in if/when we need it